### PR TITLE
Register and Render Links with Icons block in PHP

### PR DESCRIPTION
**Pull request by Gentyspun**
Ref: https://github.com/amnestywebsite/humanity-theme/issues/143

Registers and renders the Links with Icons block in PHP and accounts for block deprecations

**Steps to test**:
1.  Edit a post
2. Check the LWI block appears in the block editor
3. Insert the block, populate with content
4. Save and view the front end

To check deprecation, create the LWI block BEFORE switching to this branch and then switch branch and make sure the block does not break

Example:
http://bigbite.im/i/L8mOXX
http://bigbite.im/i/Z0r7uP

### CODE REVIEW

File: wp-content/themes/humanity-theme/includes/blocks/links-with-icons/render.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+if ( ! function_exists( 'render_links_with_icons_block' ) ) {
+	/**
+	 * Render a links with icons block
+	 *
+	 * @package Amnesty\Blocks
+	 *
+	 * @param array  $attrs the block attributes
+	 * @param string $content the block content
+	 *
+	 * @return string|null
```
Comment (jaymcp) [inline wp-content/themes/humanity-theme/includes/blocks/links-with-icons/render.php:14]:
```suggestion
	 * @return string
```

**Review by jaymcp — DISMISSED** [review]
Great work on this, thanks! I've left a few comments inline with some suggestions.

File: wp-content/themes/humanity-theme/includes/blocks/links-with-icons/register.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+if ( ! function_exists( 'register_links_with_icons_block' ) ) {
+	/**
+	 * Register the links with icons block
+	 *
+	 * @package Amnesty\Blocks
+	 *
+	 * @return void
+	 */
+	function register_links_with_icons_block() {
```
[the remark this answers is not in the document]
Comment (Gentyspun) [reply wp-content/themes/humanity-theme/includes/blocks/links-with-icons/register.php:13]: resolved

File: wp-content/themes/humanity-theme/includes/blocks/links-with-icons/render.php (hunk 1 above)
Comment (Gentyspun) [reply wp-content/themes/humanity-theme/includes/blocks/links-with-icons/render.php:14]: resolved

File: wp-content/themes/humanity-theme/includes/blocks/links-with-icons/render.php [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	 */
+	function render_links_with_icons_block( $attrs = [], $content ) {
```
[the remark this answers is not in the document]
Comment (Gentyspun) [reply wp-content/themes/humanity-theme/includes/blocks/links-with-icons/render.php:16]: resolved

File: wp-content/themes/humanity-theme/includes/blocks/links-with-icons/render.php [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+		$attrs = apply_filters( 'links_with_icons_block_attributes', $attrs );
+
+		$quantity         = $attrs['quantity'] ?? 0;
+		$orientation      = $attrs['orientation'] ?? 'horizontal';
+		$background_color = $attrs['backgroundColor'] ?? '';
+		$hide_lines       = $attrs['hideLines'] ?? false;
+		$divider_icon     = $attrs['dividerIcon'] ?? 'none';
+		$is_style_square  = $attrs['className'] ?? '';
```
[the remark this answers is not in the document]
Comment (Gentyspun) [reply wp-content/themes/humanity-theme/includes/blocks/links-with-icons/render.php:24]: resolved

File: private/src/scripts/editor/blocks/links-with-icons/deprecated.jsx [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+    default: 2,
+  },
+  hideLines: {
+    type: 'boolean',
+    default: false,
+  },
+  dividerIcon: {
+    type: 'text',
+    default: 'none',
+  },
```
[the remark this answers is not in the document]
Comment (Gentyspun) [reply private/src/scripts/editor/blocks/links-with-icons/deprecated.jsx:24]: resolved

**Review by jaymcp — COMMENTED** [review]
Thanks for addressing the feedback!

I'm seeing something a little odd on the frontend, and I'm not sure how we deal with it.
For blocks that existed before this change (i.e old ones that will be migrated), when they're brought up to date, we end up with a double wrapper:


However, for blocks that are created _after_ this change, there is no duplication of the wrapper.

Perhaps the inner block also needs PHP rendering? I'm not actually sure what the action is, here.

**Review by jaymcp — APPROVED** [review]
Thanks for the work on this!

**Pull request merged**